src/settings/storeconf.cpp

```diff
@@ -9,126 +9,165 @@
 #include <QFileDialog>
 #include <QApplication>
 
-StoreConf::StoreConf(QWidget * parent)
-    : QWidget (parent)
+StoreConf::StoreConf(QString userName)
+    : mConFile {userName += ".json"}
 {
-    conFile = getPathToConFile();
-    if (!conFile.size()) // conf.json file not exist. keep all fields with default values,
-    {                   // create conf.json with these values
+    mFields["applicationName"] = "Pair Storm";       //  default values used when json file
+    mFields["applicationVersion"] = "0.1";           //      not exist or corrupted
+    mFields["organizationName"] = "Lv-425.C++";
+    mFields["style"] = "WHITE";
+    mFields["editorFontName"] = "Consolas";
+    mFields["editorFontSize"] = "12";
+    mFields["defaultUdpPortNumber"] = "36108";
+    mFields["defaultTcpPortNumber"] = "32807";
+    mFields["userName"] = "unnamed user";
+    mFields["cppExtentions"] = ".c;.cpp;.h;.hpp;.json;.txt";
+    mCppExtentionsList << ".c" << ".cpp" << ".h" << ".hpp" << ".json" << ".txt";
+
+}
+
+void StoreConf::restoreConFile()
+{
+    checkConfDir();
+    getPathToConFile();
+    QFile file(mPathToConFile);
+    if (!file.exists()) // userName.json file not exist. keep all fields with default values,
+    {                   // create userName.json with these values
         writeJson();
         saveData();
-    } else {
+    }
+    else
+    {
         readJson();
-        parseJson();
+        if (mReadStatus)
+        {
+            parseJson();
+        }
+        else
+        {                        //  file corrupted
+            writeJson();        //      rewrite with default values
+        }
         saveData();
     }
```

Comment:
saveDate can be moved from if / else statement to avoid duplication

Reply:
done